component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -162,13 +165,19 @@ protected void processAuthenticationResponse(HttpServletRequest httpServletReque
                 String subjectAttribute;
                 if (alternativeNamePattern != null) {
                      alternativeName = getMatchedAlternativeName(cert, authenticationContext);
+                     if (X509UserStoreName != null) {
+                         alternativeName = X509UserStoreName + "/" + alternativeName;
```

Comment:
There's a Util method to prepend the domain to the username in user core. Use that instead of hardcoding the domain separator